lib/internal/http2/util.js

```diff
@@ -415,16 +417,18 @@ function mapToHeaders(map,
       if (isArray) {
         for (var k = 0; k < value.length; k++) {
           val = String(value[k]);
-          ret.push([key, val]);
+          ret += `${key}\0${val}\0`;
         }
+        count += value.length;
       } else {
         val = String(value);
-        ret.push([key, val]);
```

Comment:
A big part of the reason this was an array is because there will be a *third* value that needs to be passed in at some point... specifically, a flag that indicates whether or not the header pair should be stored in the header compression table